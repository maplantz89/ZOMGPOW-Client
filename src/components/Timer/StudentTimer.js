import React from 'react';
import './StudentTimer.css';

class StudentTimer extends React.Component {

  constructor(props){
    super(props)
    this.state = {
      timer: false,
      currMin: 0,
      currSec: '00',
    }
  }


  componentDidMount(){
    if(this.props.currTimer){
      this.handleDisplayTimer(this.props.currTimer.end)
    }
  }

  handleDisplayTimer = (endTime) => {
    this.setState({
      timer: true,
      interval: setInterval(this.displayTimer, 1000, endTime)
    })
    
  }

  displayTimer = (endTime) => {
    console.log('display timer ran')
    
    let now = Date.now()

    if(now > endTime){
      clearInterval(this.state.interval)
      this.setState({
        currMin: 0,
        currSec: '00'
      })
      return
    }

    let milliSecs = endTime - now;
    let secs = milliSecs / 1000;
    console.log('seconds', secs)
    let minutes = (secs / 60).toFixed(0)
    let displaySecs = (secs - (minutes * 60)).toFixed(0)
    if(displaySecs < 10){
      displaySecs = `0${displaySecs}`
    }

    this.setState({
      currMin: minutes,
      currSec: displaySecs
    })

  }


  render() {
    
    const {currMin, currSec} = this.state
    
    return (
      <div className='timer'>{currMin}:{currSec}</div>
    )
  }
}

export default StudentTimer;