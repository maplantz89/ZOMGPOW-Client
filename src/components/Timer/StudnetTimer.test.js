import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import StudentTimer from './StudentTimer';

describe(`StudentTimer component`, () => {
  it('renders a .StudentTimer by default', () => {
    const wrapper = shallow(<StudentTimer />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})