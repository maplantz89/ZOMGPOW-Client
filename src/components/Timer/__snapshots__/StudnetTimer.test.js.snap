// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentTimer component renders a .StudentTimer by default 1`] = `
<div>
  Timer display
</div>
`;
